pydash/utilities.py

```diff
@@ -85,6 +92,42 @@ def attempt(func, *args, **kargs):
     return ret
 
 
+def cond(*pairs):
+    """Creates a func that iterates over :attr:`pairs` and invokes the
+    corresponding function of the first predicate to return truthy.
+
+    Args:
+        pairs (list): A list of predicate-function pairs.
+
+    Returns:
+        function: Returns the new composite function.
+
+    Example:
+
+        >>> func = cond([matches({'a': 1}), constant('matches A')])
+        >>> func({'a': 1, 'b': 2})
+        'matches A'
+
+    .. versionadded:: TODO
+    """
+    for pair in pairs:
+        if len(pair) != 2:
```

Comment:
No, you should still raise the `ValueError` but what I'm saying is that `len(pair)` may fail so you need to handle that too in order to ensure that the `ValueError` is raised vs. whatever error might be thrown when you do `len(pair)`.

An alternative for this could be:

```python
try:
    assert len(pair) == 2
except Exception:
    raise ValueError(...)

if not all(map(callable, pair)):
    raise TypeError(...)
```